### ISSUE TITLE
stackoverflow on broadcast

### ISSUE THREAD

**Issue by Orbots** [chakravala/Grassmann.jl#95]
Julia-1.6.4
Grassmann-0.7.7

```
julia> basis"+++"

julia> (⋆).([1v, 1v1])
ERROR: StackOverflowError:
Stacktrace:
     [1] promote_rule(#unused#::Type{Simplex{V, G, B, T}}, #unused#::Type{S}) where {V, G, T, B, S}
       @ Grassmann C:\Users\michael.ewert\.julia\packages\Grassmann\xDWIy\src\products.jl:409
     [2] promote_type(#unused#::Type{T}, #unused#::Type{S}) where {T, S}
       @ Base .\promotion.jl:233
     [3] promote_type(::Type, ::Vararg{Type, N} where N)
       @ Leibniz C:\Users\michael.ewert\.julia\packages\Leibniz\6QRov\src\utilities.jl:22
     [4] promote_rule(#unused#::Type{Simplex{⟨+++⟩, 0, v, Simplex{⟨+++⟩, 1, v₁, Int64}}}, #unused#::Type{Simplex{⟨+++⟩, 1, v₁, Simplex{⟨+++⟩, 0, v, Int64}}})
       @ Grassmann C:\Users\michael.ewert\.julia\packages\Grassmann\xDWIy\src\products.jl:409
     [5] promote_type
       @ .\promotion.jl:233 [inlined]
     [6] promote_result(#unused#::Type, #unused#::Type, #unused#::Type{Simplex{⟨+++⟩, 0, v, Simplex{⟨+++⟩, 1, v₁, Int64}}}, #unused#::Type{Simplex{⟨+++⟩, 1, v₁, Simplex{⟨+++⟩, 0, v, Int64}}})
       @ Base .\promotion.jl:247
     [7] promote_type(#unused#::Type{T}, #unused#::Type{S}) where {T, S}
       @ Base .\promotion.jl:233
     [8] promote_type(::Type, ::Vararg{Type, N} where N)
       @ Leibniz C:\Users\michael.ewert\.julia\packages\Leibniz\6QRov\src\utilities.jl:22
--- the last 8 lines are repeated 10042 more times ---
 [80345] promote_rule(#unused#::Type{Simplex{V, G, B, T}}, #unused#::Type{S}) where {V, G, T, B, S}
       @ Grassmann C:\Users\michael.ewert\.julia\packages\Grassmann\xDWIy\src\products.jl:409
 [80346] promote_type(#unused#::Type{T}, #unused#::Type{S}) where {T, S}
       @ Base .\promotion.jl:233
 [80347] promote_type(::Type, ::Vararg{Type, N} where N)
       @ Leibniz C:\Users\michael.ewert\.julia\packages\Leibniz\6QRov\src\utilities.jl:22
 [80348] promote_rule(#unused#::Type{Simplex{⟨+++⟩, 0, v, Simplex{⟨+++⟩, 1, v₁, Int64}}}, #unused#::Type{Simplex{⟨+++⟩, 1, v₁, Simplex{⟨+++⟩, 0, v, Int64}}})
       @ Grassmann C:\Users\michael.ewert\.julia\packages\Grassmann\xDWIy\src\products.jl:409
 [80349] promote_type
       @ .\promotion.jl:233 [inlined]
 [80350] promote_result(#unused#::Type, #unused#::Type, #unused#::Type{Simplex{⟨+++⟩, 0, v, Simplex{⟨+++⟩, 1, v₁, Int64}}}, #unused#::Type{Simplex{⟨+++⟩, 1, v₁, Simplex{⟨+++⟩, 0, v, Int64}}})
       @ Base .\promotion.jl:247
 [80351] promote_type
       @ .\promotion.jl:233 [inlined]
 [80352] promote_typeof(x::Simplex{⟨+++⟩, 0, v, Int64}, xs::Simplex{⟨+++⟩, 1, v₁, Int64})
       @ Base .\promotion.jl:272
 [80353] vect(::Simplex{⟨+++⟩, 0, v, Int64}, ::Vararg{Any, N} where N)
       @ Base .\array.jl:126
```

**Comment by chakravala** [comment]
Thanks for sharing the bug report. This is a low priority bug for my work right now. If anyone has any information on how to fix this, let me know as it would make sense to fix eventually.

**Comment by chakravala** [comment]
You can get rid of the error message by deleting these lines of code https://github.com/chakravala/Grassmann.jl/blob/44d5f9db6f0608d10d76c9300c22f6aaba0a3b3e/src/products.jl#L701-L705

However, I may not be tag an official release commit until I finalize some other changes first right now.

**Comment by Orbots** [comment]
No rush.  This is more for Grassmann than myself.

**Comment by chakravala** [comment]
Issue is fixed in v0.8 of `Grassmann`, also @Orbots in v0.8 the operators `<<` and `>>` are now the conventional left and right contraction, which you requested a while back earlier. This is a breaking change.
